client/engine/engine.go

```diff
@@ -245,31 +265,102 @@ func (e *Engine) constructObjectiveFromMessage(message protocols.Message) (proto
 			message.SignedStates[0].State(),
 			*e.store.GetAddress(),
 		)
+	case strings.Contains(string(message.ObjectiveId), "Virtual"):
```

Comment:
Probably beyond the scope of this PR but this logic should probably live in the protocol package